columnflow/selection/cms/jets.py

```
@@ -0,0 +1,137 @@
+    under ``jet_veto_map``:
+
+    .. code-block:: python
+
+        cfg.x.external_files = DotDict.wrap({
+            "jet_veto_map": ("/afs/cern.ch/user/m/mfrahm/public/mirrors/jsonpog-integration-a332cfa/POG/JME/2022_Summer22EE/jetvetomaps.json.gz", "v1"), # noqa
+        })
+
+    *get_veto_map_file* can be adapted in a subclass in case it is stored differently in the external files.
+
+    documentation: https://cms-jerc.web.cern.ch/Recommendations/#jet-veto-maps
+    """
+
+    jet = events.Jet
+    muon = events.Muon[events.Muon.isPFcand]
+
+    jet_mask = (
+        (jet.pt > 15) &
+        (jet.jetId >= 2) &
+        (jet.chEmEF < 0.9) &
+        (ak.all(events.Jet.metric_table(muon) >= 0.2, axis=2))
+    )
+    # apply loose Jet puId in Run 2 to jets with pt below 50 GeV
+    if self.config_inst.campaign.x.year <= 2018:
+        jet_pu_mask = (events.Jet.puId >= 4) | (events.Jet.pt > 50)
+        jet_mask = jet_mask & jet_pu_mask
+
+    variable_map = {
+        "type": "jetvetomap",
+        "eta": jet.eta,
+        "phi": jet.phi,
+    }
+
+    inputs = [variable_map[inp.name] for inp in self.veto_map.inputs]
+
+    # apply the veto map
+    veto_map_result = ak.where(
+        jet_mask,
+        self.veto_map(*inputs),
+        -1,
+    )
+
+    # get the Jet veto mask (events containing such a jet should be vetoed)
+    veto_map_jet_mask = (veto_map_result > 0)
+
+    if "postBPix" in self.config_inst.campaign.name:
+        # in postBPix, we need to run the veto map with type=jetvetomap_bpix and subtract this from
+        # the result of the nominal jet veto map
+        raise NotImplementedError("Jet Veto Map for 2023 postBPix not implemented yet")
+
+    # add the veto map result to the events
+    events = set_ak_column(events, "Jet.veto_maps", veto_map_result)
+    results = SelectionResult(
+        steps={"jet_veto_map": ak.sum(veto_map_jet_mask, axis=1) >= 1},
+        objects={"Jet": {"veto_map_jets": veto_map_jet_mask}},
```

I don't think we should provide objects in "helper" selectors by default, but it should be a user decision. Also, since events with vetoed jets are to be discarded altogether, creating objects by default would be a bit misleading.

---

Btw, to "return" the actual per-jet mask you could also add it to the aux dict of the SelectionResult.

---

Good point, returning an aux is much better